File: apps/api/app/schemas/country.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class CountryRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    iso3: str
    iso2: str
    name: str
    region: str
    created_at: datetime
    updated_at: datetime
    observation_count: Optional[int] = None
    indicator_count: Optional[int] = None
    latest_observation_year: Optional[int] = None

File: apps/web/types/country.ts
export interface Country {
  iso3: string;
  iso2: string;
  name: string;
  region: string;
  created_at: string;
  updated_at: string;
  observation_count?: number | null;
  indicator_count?: number | null;
  latest_observation_year?: number | null;
}